contributing/google-summer-of-code-2018/README.md

```diff
@@ -2,4 +2,147 @@
 redirect_from:
   - /contributing/google-summer-of-code/
 ---
-# Google Summer of Code 2018
\ No newline at end of file
+# Google Summer of Code 2018
+
+## How to apply
+
+Interested students should email the mentor(s) directly for discussion and further information.  You can also find them, most of the time, on our community server https://open.rocket.chat/  for real-time chat.   Many of the mentors are fluent in both Portuguese and English.  
+
+If you have ideas and proposals that are not on our idea list, or if a mentor is not available, please email to gsoc+2018@rocket.chat.
+
+For some real-time help on next steps, please visit our community server Google Summer of Code 2018 discussion channel:  
+
+https://open.rocket.chat/channel/gsoc18
+
+For general discussions and information on the 2018 program, please see GSoC 2018 category on our discussion forums:
+
+https://forums.rocket.chat/c/gsoc
+
+Our application to become a 2018 GSoC organization is currently under review.  We will have more specific student application and deadline information here once (if) we are approved.
+
+## Project Ideas  (work in progress)
+
+### Global Search
+* **Mentors:** @rodrigo.nascimento
+* **Description:** Add global search to Rocket.Chet, allowing users to search across messages, rooms, files, users, etc in a single and easy to use interface. Like a MacOS Spotlight search.
+* **Desirable Skills:** Familiarity with JavaScript development, MongoDB or other search engine.
+
+### Bot2Bot Creator
```

Comment:
To be honest, this idea seems to be a conflict with our current incoming and outgoing integrations along with the ideas behind Rocket.Chat Apps.